src/authInfo.ts

```diff
@@ -444,8 +444,8 @@ export class AuthInfo extends AsyncCreatable<AuthInfo.Options> {
    *
    * @param options The options to generate the URL.
    */
-  public static getAuthorizationUrl(options: OAuth2Options & { scope: string }): string {
-    const oauth2 = new OAuth2WithVerifier(options);
+  public static getAuthorizationUrl(options: OAuth2Options & { scope?: string }, oauth2?: OAuth2WithVerifier): string {
+    oauth2 = oauth2 || new OAuth2WithVerifier(options);
```

Comment:
I think its better to not replace/reassign the parameters from the function, can we use a const here instead? (So we can maintain immutable the parameters)

Reply:
I personally think for optional params this is a fair pattern. I think the code is cleaner.